api/v1/address.go

```diff
@@ -1,55 +1,133 @@
 package api
 
 import (
+	"encoding/json"
 	"net/http"
 
+	"github.com/data-preservation-programs/spade-tenant/core"
+	"github.com/data-preservation-programs/spade-tenant/db"
 	"github.com/labstack/echo/v4"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
 )
 
-type Addresses []Address
-type AddressesMutable []AddressMutable
+type Addresses []db.Address
+type Address db.Address
 
-type Address struct {
-	AddressMutable
-	ActorID *uint `json:"actor_id,omitempty"` // TODO :swagger docs should be null not 0
+type AddressResponse struct {
+	AddressRobust    string `json:"address_robust"`
+	AddressActorID   uint   `json:"actor_id"`
+	AddressIsSigning bool   `json:"is_signing"`
 }
 
-type AddressMutable struct {
-	Address   string `json:"address"`
-	IsSigning bool   `json:"is_signing"` // true - active dealmaking from this address, false - still counts as an associated wallet
+var service *core.SpdTenantSvc
```

Comment:
hmm, having this global variable which we define is definitely not a good pattern here - need to share the state in a more structured way IMO.  Sorry for the multiple revisions on this thing, I didn't fully think through how it would work 

I think the best way to do that is the following:

1. in `v1/api.go` define a new `struct apiV1`. For now its only member value should be a pointer to the db i.e `db *gorm.DB`
2. Make a constructor for it i.e, ``. 
```go
func NewApiV1(db *gorm.DB) *apiV1 {
  return &apiV1{
    db: db
  }
}
```
3. Change `RegisterRoutes` to be a receiver function on `apiV1` i.e 
```
func (a*apiV1) RegisterRoutes(e *echo.Echo)
```
4. Change the rest of the route handlers to also be receiver functions, i.e, `func handleSetAddresses` -> `func (a *apiV1) handleSetAddresses`, now you can use `a.db` within those functions

5. Now it's the calling code (i.e, in  the top-level `main.go` to call `NewApiV1` (passing in the already-constructed DB object), then call `RegisterRoutes` on that thing, and everything should work

At this point, I think removing the `core/` subpackage actually makes sense (I know we just made it, sorry!). Instead, the top-level `main.go` has the responsibility of calling the `InitConfig()`, then constructing whatever it needs to pass into the new apiV1 thing to make it work.

Reply:
I based this off of Estuary's design of the same thing- https://github.com/application-research/estuary/blob/dev/api/v1/api.go